core/policyToken.go

```diff
@@ -232,11 +272,33 @@ func vaultHealthCheck(config *tomlConfig, httpClient *http.Client) (sCode int, e
 	}
 	defer resp.Body.Close()
 
-	lc.Info(fmt.Sprintf("Vault Health Check HTTP Status: %s (StatusCode: %s)", resp.Status, strconv.Itoa(resp.StatusCode)))
+	lc.Info(fmt.Sprintf("Vault Policy Check HTTP Status: %s (StatusCode: %s)", resp.Status, strconv.Itoa(resp.StatusCode)))
 
 	return resp.StatusCode, nil
 }
 
+// ----------------------------------------------------------
+func hashFile(policyFilePtr *string) (hashSum []byte, err error) {
+
+	inputFile, err := os.Open(*policyFilePtr)
+	if err != nil {
+		lc.Error(fmt.Sprintf("Fatal Error opening the Vault policy file: %s", err))
+		return nil, err
+	}
+	defer inputFile.Close()
+
+	hash256 := sha256.New()
+	if _, err := io.Copy(hash256, inputFile); err != nil {
+		lc.Error(fmt.Sprintf("Fatal Error calculating the SHA256 hash of the Vault policy file: %s", err))
+		return nil, err
+	}
+
+	hashSum = hash256.Sum(nil)
+	lc.Info(fmt.Sprintf("Vault policy file checksum (SHA256): %x", hashSum))
```

Comment:
Shouldn't this probably be a debug message?

Reply:
@anonymouse64 yes and no, as it is a first step into a more sophisticated comparative and enforcement process it'll probably change depending on the use cases. Will keep that in mind.

Reply:
Well even for the eventual feature of ensuring the policy file hasn't been tampered with, you don't really care what the hash is, you just care has it been tampered with, so you just would compare the hash with the expected hash and print an info message about whether it matches (or an error if it doesn't match). I don't think folks want or need to see big hashes in the log file unless they are turning on debug mode.

Reply:
@anonymouse64 Basically you have described how the feature will be. I probably thought for a minute that before it is completely implemented having the hash info could be somehow interesting. Fair enough then, I'll put this into a debug case for now.